tensorboard/examples/plugins/example_raw_scalars/tensorboard_plugin_example_raw_scalars/plugin.py

```diff
@@ -122,15 +125,15 @@ def scalars_impl(self, tag, run):
           errors.NotFoundError: if run+tag pair has no scalar data.
         """
         try:
-            tensor_events = self._multiplexer.Tensors(run, tag)
-            values = [
-                (
-                    tensor_event.wall_time,
-                    tensor_event.step,
-                    tensor_util.make_ndarray(tensor_event.tensor_proto).item(),
-                )
-                for tensor_event in tensor_events
-            ]
+            all_scalars = self._data_provider.read_scalars(
+                ctx,
+                experiment_id=experiment,
+                plugin_name=metadata.PLUGIN_NAME,
+                downsample=5000,
+                run_tag_filter=provider.RunTagFilter(runs=[run], tags=[tag]),
+            )
+            scalars = all_scalars.get(run, {}).get(tag, None)
```

Comment:
Now that these dict getters have default values, it seems the `KeyError` should never happen, but a `TypeError: 'NoneType' object is not iterable` might occur if it tries to iterate over `None`.  Can we update the docstring's `Raises: ...` and change the exception type?